core/src/main/resources/error/error-classes.json

```diff
@@ -77,6 +144,9 @@
     "message" : [ "PARTITION clause cannot contain a non-partition column name: %s" ],
     "sqlState" : "42000"
   },
+  "NO_SUCH_ELEMENT_EXCEPTION" : {
```

Comment:
We should:
- Remove the word `EXCEPTION` from the error class.
- Update the error message to better reflect what type of error this is, maybe with `No such element %s`?